### PR TITLE
Updates build to JDK 14 and addresses latest errorprone warnings

### PR DESCRIPTION
**Pull request by codefromthecrypt**

### CODE REVIEW

File: zipkin-collector/activemq/src/main/java/zipkin2/collector/activemq/LazyInit.java [hunk 1]
```diff
@@ -58,6 +58,7 @@ void close() {
     if (maybe != null) result.close();
   }
 
+  @SuppressWarnings("EmptyCatch")
```
Comment (anuraaga) [inline zipkin-collector/activemq/src/main/java/zipkin2/collector/activemq/LazyInit.java]: This is fine too though I think `// Ignore` in the catch block is fine too and probably what this check is intending, just making sure you didn't accidentally forget

File: zipkin-collector/rabbitmq/src/main/java/zipkin2/collector/rabbitmq/RabbitMQCollector.java [hunk 2]
```diff
@@ -257,12 +257,13 @@ public void handleDelivery(String tag, Envelope envelope, BasicProperties props,
     for (int i = 0; i < addresses.size(); i++) {
       String[] splitAddress = addresses.get(i).split(":", 100);
       String host = splitAddress[0];
-      Integer port = null;
+      int port = -1;
       try {
         if (splitAddress.length == 2) port = Integer.parseInt(splitAddress[1]);
       } catch (NumberFormatException ignore) {
+        port = 0;
```
Comment (anuraaga) [inline zipkin-collector/rabbitmq/src/main/java/zipkin2/collector/rabbitmq/RabbitMQCollector.java]: Guess you might have added this for the same check but consider a comment instead

File: zipkin-server/src/main/java/zipkin2/server/internal/health/ZipkinHealthController.java [hunk 3]
```diff
@@ -48,6 +48,7 @@
     return health(ctx, MediaType.JSON_UTF_8);
   }
 
+  @SuppressWarnings("FutureReturnValueIgnored")
```
Comment (anuraaga) [inline zipkin-server/src/main/java/zipkin2/server/internal/health/ZipkinHealthController.java:51]: Ah too bad this annoying pre-Java8 check is still around

File: zipkin-storage/cassandra-v1/src/main/java/zipkin2/storage/cassandra/v1/CassandraUtil.java [hunk 4]
```diff
@@ -142,4 +144,13 @@ public String toString() {
       return "SortTraceIdsByDescTimestamp";
     }
   }
+
+  @SuppressWarnings("JdkObsolete")
+  static List<Date> getDays(long endTs, @Nullable Long lookback) {
+    List<Date> result = new ArrayList<>();
+    for (Long epochMillis : DateUtil.epochDays(endTs, lookback)) {
```
Comment (anuraaga) [inline zipkin-storage/cassandra-v1/src/main/java/zipkin2/storage/cassandra/v1/CassandraUtil.java]: Accidental box?

File: zipkin-storage/elasticsearch/src/main/java/zipkin2/elasticsearch/ElasticsearchVersion.java [hunk 5]
```diff
@@ -39,6 +39,8 @@ float get(HttpCall.Factory callFactory) throws IOException {
     try {
       if (enterPath(parser, "version", "number") != null) version = parser.getText();
     } catch (RuntimeException | IOException possiblyParseException) {
+      throw new IllegalArgumentException(
```
Comment (anuraaga) [inline zipkin-storage/elasticsearch/src/main/java/zipkin2/elasticsearch/ElasticsearchVersion.java]: On phone so can't expand this file to see below but if it's duping the next line, commet instead is fine

**Review by anuraaga — APPROVED** [review]

File: zipkin-collector/activemq/src/main/java/zipkin2/collector/activemq/LazyInit.java (hunk 1 above)
Comment (codefromthecrypt) [reply zipkin-collector/activemq/src/main/java/zipkin2/collector/activemq/LazyInit.java]: yeah generally I am just trying to get the hugely annoying errorprone messages to go away one way or another. if an empty comment or // ignore works that works for me.

File: zipkin-storage/cassandra-v1/src/main/java/zipkin2/storage/cassandra/v1/CassandraUtil.java (hunk 4 above)
Comment (codefromthecrypt) [reply zipkin-storage/cassandra-v1/src/main/java/zipkin2/storage/cassandra/v1/CassandraUtil.java]: well it is a List<Long> but sure no prob making it eagerly unboxed.

**Comment by codefromthecrypt** [comment]
@anuraaga great tip on the comment in empty catch.. wish I thought of it, but glad you did :)

**Comment by anuraaga** [comment]
LGTM

I think we had the same reaction but will admit to being better at getting in the shoes of the oppressor ;)

**Pull request merged**